### PR TITLE
3225 changes to history

### PR DESCRIPTION
**Pull request by denis-s-oms**
1. New item name for consultant fee;
2. Skipped changes for user's field `disabled_at`;
3. Extended item name for factor, floor, dpo, js targeting;
4. Saving changes for `Created` and `Deleted` actions as changes from `null` to `value` and vice versa;
5. Added column `demand_partner_id`, saving it for dpo changing;
6. Resolved issue with domain's `integration_type`;
7. Changed history module in order to allow saving history for workers;
8. Saving history for `bulk/dpo` and `bulk/factor`;

### CODE REVIEW

File: core/bulk/dpo.go [hunk 1]
```diff
@@ -44,18 +45,36 @@ func BulkInsertDPO(ctx context.Context, requests []core.DPOUpdateRequest) error
...
 	for i, chunk := range chunks {
-		dpos := prepareDPO(chunk, demandPartners)
+		dpos, oldDpos := prepareDPO(ctx, chunk, demandPartners)
 		if err := bulkInsertDPO(ctx, tx, dpos); err != nil {
 			log.Error().Err(err).Msgf("failed to process dpos bulk update for chunk %d", i)
-			return fmt.Errorf("failed to process dpos bulk update for chunk %d: %w", i, err)
+			return nil, nil, fmt.Errorf("failed to process dpos bulk update for chunk %d: %w", i, err)
+		}
+
+		oldMods = append(oldMods, oldDpos...)
```
Comment (finkiel) [inline core/bulk/dpo.go:71]: I would wrap those 3 lines as a function (for example handleHistory)- it's hard to understand what is oldMods, newMods  in the context of the method handleDpoRuleTable

File: core/bulk/dpo.go [hunk 2]
```diff
@@ -104,11 +124,35 @@ func prepareDPO(chunk []core.DPOUpdateRequest, demandPartners map[string]struct{
...
+		ids = append(ids, dpo.RuleID)
+		dpos = append(dpos, dpo)
+	}
+
+	oldMods, err := models.DpoRules(models.DpoRuleWhere.RuleID.IN(ids)).All(ctx, bcdb.DB())
+	if err != nil {
+		return nil, nil
+	}
+
+	m := make(map[string]*models.DpoRule)
```
Comment (finkiel) [inline core/bulk/dpo.go:140]: I think that you should create a func from lines 135-153:
1. will give better understanding that this part is for the history
2. the method is too long and it should be break into smaller part (at least in my opinion :) )

File: core/bulk/dpo.go (hunk 1 above)
Comment (denis-s-oms) [reply core/bulk/dpo.go:80]: Not sure that these three lines are enough to wrap them as function. Added comment to make it more clear.

File: core/bulk/dpo.go (hunk 2 above)
Comment (denis-s-oms) [reply core/bulk/dpo.go:155]: Done.

File: core/bulk/factor.go [hunk 3]
```diff
@@ -119,16 +112,31 @@ func prepareMetaDataWithFactors(ctx context.Context, pubDomains map[string]struc
...
-		factors := createFactorsData(chunk, pubDomains)
+		factors, oldFactors := createFactorsData(ctx, chunk, pubDomains)
 
 		if err := bulkInsertFactors(ctx, tx, factors); err != nil {
 			log.Error().Err(err).Msgf("failed to process factor bulk update for chunk %d", i)
-			return fmt.Errorf("failed to process factor bulk update for chunk %d: %w", i, err)
+			return nil, nil, fmt.Errorf("failed to process factor bulk update for chunk %d: %w", i, err)
+		}
+
+		oldMods = append(oldMods, oldFactors...)
```
[the remark this answers is not in the document]
Comment (denis-s-oms) [reply core/bulk/factor.go:141]: Not sure that these three lines are enough to wrap them as function. Added comment to make it more clear.

File: core/bulk/factor.go [hunk 4]
```diff
@@ -160,19 +169,40 @@ func createFactorsData(chunk []FactorUpdateRequest, pubDomain map[string]struct{
...
 
 		factor.RuleId = factor.GetRuleID()
-		factors = append(factors, *factor.ToModel())
+		ids = append(ids, factor.RuleId)
+		factors = append(factors, factor.ToModel())
 		pubDomain[data.Publisher+":"+data.Domain] = struct{}{}
 	}
-	return factors
+
+	oldMods, err := models.Factors(models.FactorWhere.RuleID.IN(ids)).All(ctx, bcdb.DB())
```
[the remark this answers is not in the document]
Comment (denis-s-oms) [reply core/bulk/factor.go:193]: Done.

**Review by finkiel — APPROVED** [review]

**Pull request merged**